### ISSUE TITLE
object has no attribute 'item_inventory_count'

### ISSUE THREAD

**Issue by cbogithub** [PokemonGoF/PokemonGo-Bot#4237]
```
Traceback (most recent call last):
  File "pokecli.py", line 595, in <module>
    main()
  File "pokecli.py", line 105, in main
    bot.tick()
  File "P:\pokemon go\PokemonGo-Bot\pokemongo_bot\__init__.py", line 532, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "P:\pokemon go\PokemonGo-Bot\pokemongo_bot\cell_workers\evolve_pokemon.py", line 27, in work
    if not self._should_run():
  File "P:\pokemon go\PokemonGo-Bot\pokemongo_bot\cell_workers\evolve_pokemon.py", line 49, in _should_run
    lucky_egg_count = self.bot.item_inventory_count(Item.ITEM_LUCKY_EGG.value)
AttributeError: 'PokemonGoBot' object has no attribute 'item_inventory_count'
```

config 

```
{
    "auth_service": "ptc",
    "username": "nnn",
    "password": "bbbbb",
    "location": "nnnn",
    "gmapkey": "vvvv",
    "tasks": [{
        "type": "HandleSoftBan"
    }, {
        "type": "CollectLevelUpReward"
    }, 
	
	  {
		"type": "RandomPause",
        "config": {
		  "enabled": false,
		  "min_duration": "00:00:10",
		  "max_duration": "00:10:00",
		  "min_interval": "00:10:00",
		  "max_interval": "02:00:00"
        }
	  },	
	
	
	
	{
        "type": "IncubateEggs",
        "config": {
            "longer_eggs_first": true
        }
    }, {
        "type": "RecycleItems",
        "min_empty_space": 10,
        "config": {
            "item_filter": {
                "Pokeball": {
                    "keep": 50
                },
                "Greatball": {
                    "keep": 100
                },
                "Ultraball": {
                    "keep": 50
                },
                "Potion": {
                    "keep": 0
                },
                "Super Potion": {
                    "keep": 0
                },
                "Hyper Potion": {
                    "keep": 0
                },
                "Revive": {
                    "keep": 0
                },
                "Max Potion": {
                    "keep": 50
                },
                "Max Revive": {
                    "keep": 50
                },
                "Razz Berry": {
                    "keep": 50
                }
            }
        }
    }, {
        "type": "EvolvePokemon",
        "config": {
            "evolve_all": "Bulbasaur,Charmander,Squirtle,Caterpie,Weedle,Pidgey,Rattata,Spearow,Ekans,Pikachu,Sandshrew,Nidoran F, Nidoran M,Clefable,Vulpix,Jigglypuff,Zubat,Oddish,Paras,Venonat,Diglett,Meowth,Poliwag,Abra,Tentacool,Geodude,Ponyta,Slowpoke,Magnemite,Doduo,Seel,Gastly,Drowzee,Krabby,Voltorb,Cubone,Koffing,Rhyhorn,Horsea,Staryu,Omanyte,Kabuto",
            "first_evolve_by": "iv",
            "evolve_above_cp": 0,
            "evolve_above_iv": 0,
            "logic": "and",
            "evolve_speed": 1,
            "use_lucky_egg": true
        }
    }, 
	{
        "type": "NicknamePokemon",
        "config": {
            "nickname_template": "{name:.8s}-{iv_pct}",
            "dont_nickname_favorite": true
        }
    }, 
	{
        "type": "TransferPokemon"
    },
	{
        "type": "MoveToMapPokemon",
        "config": {
            "address": "http://localhost:5000",
            "mode": "priority",
            "prioritize_vips": true,
            "min_time": 15,
            "max_distance": 1,
            "snipe": true,
            "min_ball": 30,
            "snipe_high_prio_only": true,
            "snipe_high_prio_threshold": 1,
            "update_map": true,
            "catch": {
                "Mewtwo": 990,
                "Mew": 980,
                "Moltres": 970,
                "Zapdos": 960,
                "Articuno": 950,

                "Snorlax": 940,
                "Lapras": 930,
                "Dragonite": 920,
                "Dragonair": 910,
                "Dratini": 900,

                "Vaporeon": 890,
                "Eevee": 885,
                "Arcanine": 880,
                "Growlithe": 875,
                "Exeggutor": 870,
                "Exeggcute": 865,
                "Gyarados": 860,
                "Magikarp": 855,
                "Muk": 850,
                "Grimer": 845,

                "Machamp": 790,
                "Machop": 797,
                "Machoke": 785,
                "Victreebel": 780,
                "Weepinbell": 787,
                "Bellsprout": 775,
                "Cloyster": 770,
                "Shellder": 765,
                "Golduck": 750,
                "Psyduck": 745,

                "Chansey": 100
            }
        }
    },
	{
        "type": "CatchVisiblePokemon"
    }, {
        "type": "CatchLuredPokemon"
    }, {
        "type": "SpinFort"
    }, {
        "type": "MoveToFort",
        "config": {
            "lure_attraction": true,
            "lure_max_distance": 2000
        }
    }, {
        "type": "FollowSpiral",
        "config": {
            "diameter": 5,
            "step_size": 100
        }
    }],
    "map_object_cache_time": 5,
    "forts": {
        "avoid_circles": true,
        "max_circle_size": 100,
        "cache_recent_forts": true
    },
    "min_ultraball_to_keep": 20,
    "catch_throw_parameters": {
        "excellent_rate": 0.11,
        "great_rate": 0.55,
        "nice_rate": 0.25,
        "normal_rate": 0.09,
        "spin_success_rate": 0.67
    },
    "websocket_server": true,

	
	
    "pokemon_bag": {
      "// if 'show_at_start' is true, it will log all the pokemons in the bag (not eggs) at bot start": {},
      "show_at_start": true,
      "// if 'show_count' is true, it will show the amount of each pokemon (minimum 1)": {},
      "show_count": false,
      "// 'pokemon_info' parameter define which info to show for each pokemon": {},
      "// the available options are": {},
      "// ['cp', 'iv_ads', 'iv_pct', 'ivcp', 'ncp', 'level', 'hp', 'moveset', 'dps']": {},
      "pokemon_info": ["cp", "iv_pct"]
    },	
	
    "walk_max": 50.16,
    "walk_min": 5.16,
    "debug": false,
    "test": false,
    "health_record": false,
    "location_cache": true,
    "distance_unit": "km",
    "reconnecting_timeout": 15,
    "logging_color": true,
    "daily_catch_limit": 800,
    "catch": {
        "any": {
            "catch_above_cp": 0,
            "catch_above_iv": 0,
            "logic": "or"
        }
    },
	
	
    "catch_simulation": {
	    "flee_count": 3,
	    "flee_duration": 2,
	    "catch_wait_min": 2,
	    "catch_wait_max": 6,
	    "berry_wait_min": 2,
	    "berry_wait_max": 3,
	    "changeball_wait_min": 2,
	    "changeball_wait_max": 3
	  },	
	
	
    "release": {
        "//100 / 97.7 / 95.5 / 93.3 /  91.1 / 88.8 / 86.6 / 84.4 / 82.2 / 80.0 / 77.7 / 75.5 / 73.3 / 71.1": {},
        "any": {
            "release_below_cp": 0,
            "release_below_iv": 0.99,
            "logic": "or"
        },

        "Snorlax": {
            "never_release": true
        },
        "Lapras": {
            "never_release": true
        },
        "Dragonite": {
            "never_release": true
        },
        "Dragonair": {
            "keep_best_iv": 3
        },
        "Dratini": {
            "keep_best_iv": 3
        },

        "Vaporeon": {
            "never_release": true
        },
        "Eevee": {
            "keep_best_iv": 3
        },
        "Arcanine": {
            "never_release": true
        },
        "Growlithe": {
            "keep_best_iv": 3
        },
        "Exeggutor": {
            "never_release": true
        },
        "Exeggcute": {
            "keep_best_iv": 3
        },
        "Gyarados": {
            "never_release": true
        },
        "Magikarp": {
            "keep_best_iv": 3
        },
        "Muk": {
            "never_release": true
        },
        "Grimer": {
            "keep_best_iv": 3
        },

        "Machamp": {
            "never_release": true
        },
        "Machoke": {
            "keep_best_iv": 3
        },
        "Machop": {
            "keep_best_iv": 3
        },
        "Victreebel": {
            "never_release": true
        },
        "Weepinbell": {
            "keep_best_iv": 3
        },
        "Bellsprout": {
            "keep_best_iv": 3
        },
        "Cloyster": {
            "never_release": true
        },
        "Shellder": {
            "keep_best_iv": 3
        },
        "Golduck": {
            "release_below_iv": 0.77
        },
        "Psyduck": {
            "release_below_iv": 0.77
        },

        "Mew": {
            "never_release": true
        },
        "Mewtwo": {
            "never_release": true
        },
        "Articuno": {
            "never_release": true
        },
        "Zapdos": {
            "never_release": true
        },
        "Moltres": {
            "never_release": true
        },
        "Chansey": {
            "keep_best_iv": 3
        },

        "Caterpie": {
            "always_release": true
        },
        "Metapod": {
            "always_release": true
        },
        "Butterfree": {
            "always_release": true
        },
        "Weedle": {
            "always_release": true
        },
        "Kakuna": {
            "always_release": true
        },
        "Beedrill": {
            "always_release": true
        },
        "Pidgey": {
            "always_release": true
        },
        "Pidgeotto": {
            "always_release": true
        },
        "Pidgeot": {
            "always_release": true
        },
        "Rattata": {
            "always_release": true
        },
        "Raticate": {
            "always_release": true
        },
        "Spearow": {
            "always_release": true
        },
        "Fearow": {
            "always_release": true
        },
        "Ekans": {
            "always_release": true
        },
        "Arbok": {
            "always_release": true
        },
        "Sandshrew": {
            "always_release": true
        },
        "Sandslash": {
            "always_release": true
        },
        "Vulpix": {
            "always_release": true
        },
        "Ninetails": {
            "always_release": true
        },
        "Zubat": {
            "always_release": true
        },
        "Golbat": {
            "always_release": true
        },
        "Paras": {
            "always_release": true
        },
        "Parasect": {
            "always_release": true
        },
        "Venonat": {
            "always_release": true
        },
        "Venomoth": {
            "always_release": true
        },
        "Diglett": {
            "always_release": true
        },
        "Dugtrio": {
            "always_release": true
        },
        "Meowth": {
            "always_release": true
        },
        "Persian": {
            "always_release": true
        },
        "Mankey": {
            "always_release": true
        },
        "Primeape": {
            "always_release": true
        },
        "Ponyta": {
            "always_release": true
        },
        "Rapidash": {
            "always_release": true
        },
        "Doduo": {
            "always_release": true
        },
        "Dodrio": {
            "always_release": true
        },
        "Drowzee": {
            "always_release": true
        },
        "Hypno": {
            "always_release": true
        },
        "Krabby": {
            "always_release": true
        },
        "Kingler": {
            "always_release": true
        },
        "Goldeen": {
            "always_release": true
        },
        "Seaking": {
            "always_release": true
        }
    },
    "vips": {
        "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {
            "catch_above_cp": 0,
            "catch_above_iv": 0.99,
            "logic": "and"
        },
        "Mewtwo": {},
        "Mew": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},
        "Snorlax": {},
        "Lapras": {},
        "Dragonite": {},
        "Dragonair": {},
        "Dratini": {},
        "Vaporeon": {},
        "Arcanine": {},
        "Exeggutor": {},
        "Gyarados": {},
        "Muk": {},
        "Grimer": {},
        "Machamp": {},
        "Victreebel": {},
        "Cloyster": {}
    }

}
```

**Comment by Arlion** [comment]
I think the line:
"min_ultraball_to_keep": 20,

while proper syntax is made redundant by the 

` "config": {
            "item_filter": {
                "Pokeball": {
                    "keep": 50
                },
                "Greatball": {
                    "keep": 100
                },
                "Ultraball": {
                    "keep": 50
                },
                "Potion": {
                    "keep": 0
                },
                "Super Potion": {
                    "keep": 0
                },
                "Hyper Potion": {
                    "keep": 0
                },
                "Revive": {
                    "keep": 0
                },
                "Max Potion": {
                    "keep": 50
                },
                "Max Revive": {
                    "keep": 50
                },
                "Razz Berry": {
                    "keep": 50`

section... Try removing it and updating the ultraball section to 20 and see if that gets you where you want to go.

**Comment by cbogithub** [comment]
i think it erorr :count lucky egg
`lucky_egg_count = self.bot.item_inventory_count(Item.ITEM_LUCKY_EGG.value)`

**Comment by BriceSD** [comment]
Fixed in #4238

**Comment by RedViper9** [comment]
Should be fixed as of a58d43720610557e84cbf51cdca9fa2c69b46912
